Failing racing condition if built with -j > 1

Fails to build if `make -jN` is bigger than one. Probably due to a racing condition.

Full build log: https://paste.ubuntu.com/p/pPJThD3w7M/
```
g++ -Wl,-O1 -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -o cpptest cpptest.o   -L/home/vanyasem/Projects/AUR/_Unity8/_Unity8-Arch/gsettings-qt-git/src/gsettings-qt/tests/../src -lgsettings-qt -lQt5Test -lQt5Core -lpthread 
/usr/bin/ld: cannot find -lgsettings-qt
collect2: error: ld returned 1 exit status
```

Another instance:
```
g++ -Wl,--as-needed -Wl,-O1 -shared -o libGSettingsQmlPlugin.so plugin.o gsettings-qml.o moc_plugin.o moc_gsettings-qml.o  -L../src -lgsettings-qt -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl /usr/lib/libQt5Qml.so /usr/lib/libQt5Network.so /usr/lib/libQt5Core.so -lpthread    
make[1]: Leaving directory '/home/pmos/build/src/gsettings-qt-c8ecb2bf269f4ed0a4477bad262751d77ae6bb72/tests'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.3.0/../../../../x86_64-alpine-linux-musl/bin/ld: cannot find -lgsettings-qt
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile.gsettings-qt:169: libGSettingsQmlPlugin.so] Error 1
```

--- a/gsettings-qt.pro
+++ b/gsettings-qt.pro
@@ -1,2 +1,14 @@
 TEMPLATE = subdirs
-SUBDIRS += src/gsettings-qt.pro GSettings/gsettings-qt.pro tests/tests.pro tests/cpptest.pro
+SUBDIRS += \
+    libgsettings-qt \
+    gsettings-qt \
+    tests \
+    cpptest
+
+libgsettings-qt.file = src/gsettings-qt.pro
+gsettings-qt.file = GSettings/gsettings-qt.pro
+gsettings-qt.depends = libgsettings-qt
+
+tests.file = tests/tests.pro
+cpptest.file = tests/cpptest.pro
+cpptest.depends = libgsettings-qt

